### PR TITLE
Testing The Controllers

### PR DESCRIPTION
**Pull request by NickEmma**
# Hi there 👋🏻

**Thanks for taking the time to review the project. Here are some things important to note about it:**

## Changes implemented in the "Controllers Spec" branch

- ✅ Create Request Spec files for each controller
- ✅ Check if the response was correct for each action
- ✅ Check if each action rendered the proper template
- ✅ Check if the response body included defined placeholders

# Final notes 📔

**Is expected that this project meets the requirements. Please let me know if anything is left to be done or can be improved. Again, thanks for the review.**

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1 @@
-<h1> Hello views </h1>
\ No newline at end of file
+<h1> Here is a list of users</h1>
```
Comment (mikethreels) [inline app/views/users/index.html.erb:1]: - [ ] in this view you should loop through every user and display every photo, name and post counter

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,2 +1 @@
-<h1> Hello views </h1>
-<p> this is awesome </p>
\ No newline at end of file
+<h1> Here is information about a given user</h1>
```
Comment (mikethreels) [inline app/views/users/show.html.erb:1]: - [ ] in this view u should use #recent_post method to render the user's 3 most recent post.

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1 @@
-<h2> This is cool </h2>
\ No newline at end of file
+<h2> Here is a list of posts for a given user </h2>
```
Comment (mikethreels) [inline app/views/posts/index.html.erb:1]: - [ ] you should use post.recent_comments to render 5 most recent comments in the app/views/posts/index.html.erb page

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Techie, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 1 above)
Comment (NickEmma) [reply app/views/users/index.html.erb:1]: @mikethreels noted but it is not required.

File: app/views/users/show.html.erb (hunk 2 above)
Comment (NickEmma) [reply app/views/users/show.html.erb:1]: @mikethreels i don't understand what you mean by to get the recent 3 posts

**Review by mikethreels — APPROVED** [review]
Hi Techie, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- linters are passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**